fedora_messaging/config.py

```diff
@@ -393,9 +356,56 @@ def setup_logging(self):
             self.load_config()
         logging.config.dictConfig(self['log_config'])
 
-    def load_config(self, filename=None):
+    def _validate(self):
+        """
+        Perform checks on the configuration to assert its validity
+
+        Raises:
+            ConfigurationException: If the configuration is invalid.
+        """
+        for key in self:
+            if key not in DEFAULTS:
+                raise exceptions.ConfigurationException(
+                    'Unknown configuration key "{}"! Valid configuration keys are'
+                    ' {}'.format(key, list(DEFAULTS.keys())))
```

Comment:
I would love to be able to do this in Bodhi's config too, but Bodhi has *dynamic* config names. Ugh.